### DISCUSSION TITLE
Suspense mode question

### DISCUSSION THREAD

**Discussion by lucas-garrido** [vercel/swr#1666] [Q&A]
hello, i don't uderstand the behaviour of suspense mode.
i have something like this in my layout 
```
        <Suspense fallback={<LoadingPage />}>
                  {children}
        </Suspense>
```
![image](https://user-images.githubusercontent.com/48165310/143271126-da06aa30-022b-4b99-8e13-d8fbec26e917.png)

my problem is that the loading page only appear in first render.

the first time i render my dashboard page the loader appears but if i go on another page and come back to dashboard the loader is not showing anymore.
is this the normal behaviour ?

**Comment by lucas-garrido** [comment]
i understand that after first render data are cached but what if i want to show loading page every time data are refreshed ?
Edit : 
I fixed my problem by clearing the cache in componentUnmount event but i think adding a no cache option would be very useful with suspense

**Reply by shuding** [reply]
This is one of the benefits of stale-while-revalidate, mind sharing why you want to show the loading status every time?

**Reply by lucas-garrido** [reply]
- components that need to initial render with the correct data. (example: react hook form default values or react select default value)
- edition form, you want to show it once data are loaded and fresh

**Comment by anothertempore** [comment]
`useSWRImmutable` is disabled auto revalidation by design, you can try `useSWR` instead: https://swr.vercel.app/docs/revalidation#disable-automatic-revalidations
